src/pybind/mgr/dashboard/frontend/src/app/shared/directives/update-scope.directive.ts

```diff
@@ -0,0 +1,34 @@
+import { AfterViewInit, Directive, ElementRef, Input, Optional } from '@angular/core';
+import { Permissions } from '../models/permissions';
+import { AuthStorageService } from '../services/auth-storage.service';
+import { PermissionScopeDirective } from './permission-scope.directive';
+
+@Directive({
+  selector: 'input:not([cdNoUpdateScope]), select:not([cdNoUpdateScope]), [cdUpdateScope]'
+})
+export class UpdateScopeDirective implements AfterViewInit {
+  private permissions: Permissions;
+  private service_name: keyof Permissions;
+
+  @Input() disabled: boolean;
+
+  constructor(
+    @Optional() private scope: PermissionScopeDirective,
+    private authStorageService: AuthStorageService,
+    private elementRef: ElementRef
+  ) {
+    this.permissions = this.authStorageService.getPermissions();
+  }
+
+  ngAfterViewInit() {
+    if (this.scope !== null) {
+      this.service_name = this.scope.cdScope;
+    }
+    if (
+      this.service_name !== undefined &&
+      (this.disabled || !this.permissions[this.service_name].update)
```

Comment:
```suggestion
      (!this.permissions[this.service_name].update)
```

Reply:
> Do we plan to use this directives in any other forms?
In all the forms where we are using this directives we are always applying it to all inputs.
In the current code it's not even possible to specify which input should be disabled or not, or select a different scope.

@tspmelo I've been thinking/trying for a while to remove the explicit invocation of Auth service in every component constructor, decouple components from authorization, and replace it with a directive. That should simplify the coding & specially the testing.

The issue with applying this to all inputs is more about enabling them rather than disabling: there are cases where some inputs have to be disabled for other reasons (e.g.: flag rules at RBD image creation form), so if you simply enable ALL of them, you'll break those inner rules.

> In the current code it's not even possible to specify which input should be disabled or not, or select a different scope.

Yes, currently this doesn't support multi-scope (that's a bonus feature, once this is in place), but the conditional disable should be working. Isn't it, @nizamial09 ?

Reply:
@epuertat @tspmelo The conditional disable is working fine.